packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -415,3 +416,90 @@ export async function getWorkerdCompatibilityDate() {
 
 	return workerdCompatibilityDate;
 }
+
+export async function installWorkersTypes(ctx: PagesGeneratorContext) {
+	const { npm } = detectPackageManager();
+
+	await installPackages(["@cloudflare/workers-types"], {
+		dev: true,
+		startText: `Installing @cloudflare/workers-types`,
+		doneText: `${brandColor("installed")} ${dim(`via ${npm}`)}`,
+	});
+	await addWorkersTypesToTsConfig(ctx);
+}
+
+// @cloudflare/workers-types are versioned by compatibility dates, so we must look
+// up the latest entrypoint from the installed dependency on disk.
+// See also https://github.com/cloudflare/workerd/tree/main/npm/workers-types#compatibility-dates
+export function getLatestTypesEntrypoint(ctx: PagesGeneratorContext) {
+	const workersTypesPath = resolve(
+		ctx.project.path,
+		"node_modules",
+		"@cloudflare",
+		"workers-types"
+	);
+
+	try {
+		const entrypoints = readdirSync(workersTypesPath);
+
+		const sorted = entrypoints
+			.filter((filename) => filename.match(/(\d{4})-(\d{2})-(\d{2})/))
+			.sort()
+			.reverse();
+
+		if (sorted.length === 0) {
+			return null;
+		}
+
+		return sorted[0];
+	} catch (error) {
+		return null;
+	}
+}
+
+export async function addWorkersTypesToTsConfig(ctx: PagesGeneratorContext) {
+	const tsconfigPath = join(ctx.project.path, "tsconfig.json");
+	if (!existsSync(tsconfigPath)) {
+		return;
+	}
+
+	const s = spinner();
+	s.start(`Adding latest types to \`tsconfig.json\``);
+
+	const tsconfig = readFile(tsconfigPath);
+	const entrypointVersion = getLatestTypesEntrypoint(ctx);
+	if (entrypointVersion === null) {
+		s.stop(
+			`${brandColor(
+				"skipped"
+			)} couldn't find latest compatible version of @cloudflare/workers-types`
+		);
+		return;
+	}
+
+	const typesEntrypoint = `@cloudflare/workers-types/${entrypointVersion}`;
+
+	let updated = tsconfig;
+
+	if (tsconfig.includes("@cloudflare/workers-types")) {
+		updated = tsconfig.replace("@cloudflare/workers-types", typesEntrypoint);
+	} else {
+		try {
+			// Note: this simple implementation doesn't handle tsconfigs containing comments
+			//       (as it is not needed for the existing use cases)
```

Comment:
I implemented this quick solution just to get things working with Next.js, before merging I should explore if we can also, without too much complications, handle jsonc tsconfigs

Reply:
Updating jsonc files doesn't seem very simple, I can see that in wrangler it was already considered not worth it: https://github.com/cloudflare/workers-sdk/blob/ccf14dda78c33446b9d35014458ce5d33cb04ae2/packages/wrangler/src/init.ts#L462-L465

Here as well a generic solution might end up too complex and not worth it...

I did try to experiment a bit with [json5](https://www.npmjs.com/package/json5) here but mixed success (the tsconfig can get parsed by during the parsing the comments are lost)

I think would keep this as is for now (since currently it is good enough for Next.js) and improve things later if need be 🤔